src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/tts_is_ready_request_test.cc

```diff
@@ -0,0 +1,254 @@
+                CreateModuleInfoSO(hmi_apis::FunctionID::TTS_GetLanguage, _))
+        .WillOnce(Return(language));
+    EXPECT_CALL(mock_hmi_capabilities_, set_handle_response_for(*language));
+    EXPECT_CALL(mock_rpc_service_, ManageHMICommand(language, _));
+
+    smart_objects::SmartObjectSPtr support_language =
+        std::make_shared<smart_objects::SmartObject>(
+            smart_objects::SmartType_Map);
+    EXPECT_CALL(
+        mock_message_helper_,
+        CreateModuleInfoSO(hmi_apis::FunctionID::TTS_GetSupportedLanguages, _))
+        .WillOnce(Return(support_language));
+    EXPECT_CALL(mock_rpc_service_, ManageHMICommand(support_language, _));
+
+    smart_objects::SmartObjectSPtr capabilities =
+        std::make_shared<smart_objects::SmartObject>(
+            smart_objects::SmartType_Map);
+    EXPECT_CALL(
+        mock_message_helper_,
+        CreateModuleInfoSO(hmi_apis::FunctionID::TTS_GetCapabilities, _))
+        .WillOnce(Return(capabilities));
+    EXPECT_CALL(mock_rpc_service_, ManageHMICommand(capabilities, _));
+  }
+
+  void PrepareEvent(const bool message_contains_param,
+                    const bool is_tts_cooperating_available,
+                    Event& out_event) {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    if (message_contains_param) {
+      (*msg)[am::strings::msg_params][am::strings::available] =
+          is_tts_cooperating_available;
+    }
+    out_event.set_smart_object(*msg);
+  }
+
+  void HMICapabilitiesExpectations() {
+    std::set<hmi_apis::FunctionID::eType> interfaces_to_update{
+        hmi_apis::FunctionID::TTS_GetLanguage,
+        hmi_apis::FunctionID::TTS_GetSupportedLanguages,
+        hmi_apis::FunctionID::TTS_GetCapabilities};
+    EXPECT_CALL(mock_hmi_capabilities_, GetRequestsRequiredForCapabilities())
+        .WillOnce(Return(interfaces_to_update));
+  }
+
+  TTSIsReadyRequestPtr command_;
+};
+
+TEST_F(TTSIsReadyRequestTest, RUN_SendRequest_SUCCESS) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[am::strings::msg_params][am::strings::number] = kStrNumber;
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+
+  RequestToHMIPtr command(CreateCommand<TTSIsReadyRequest>(command_msg));
+  EXPECT_CALL(mock_rpc_service_, SendMessageToHMI(command_msg));
+  ASSERT_TRUE(command->Init());
+
+  command->Run();
+
+  EXPECT_EQ(CommandImpl::hmi_protocol_type_,
+            (*command_msg)[strings::params][strings::protocol_type].asInt());
+  EXPECT_EQ(CommandImpl::protocol_version_,
+            (*command_msg)[strings::params][strings::protocol_version].asInt());
+}
+
+TEST_F(TTSIsReadyRequestTest,
+       Run_NoKeyAvailableInMessage_HmiInterfacesIgnored_CacheIsAbsent) {
+  const bool is_tts_cooperating_available = false;
+  const bool should_message_be_sent = true;
+  const bool message_contains_param = false;
+  const am::HmiInterfaces::InterfaceState state =
```

Comment:
@ydementieiev const auto?

Reply:
@AByzhynar changed to [auto](https://github.com/LuxoftSDL/sdl_core/pull/35/commits/d2bcee64dd404ebb5863337c9877ee774b0bde93)